bip-hashrate-escrows.mediawiki

```diff
@@ -0,0 +1,469 @@
+|-
+| 10
+| "CTIP" -- Part 1 "TxID"*
+| 32
+| hex
+| The CTIP, or "Critical (TxID, Index) Pair" is a variable for keeping track of where the escrow's money is (ie, which member of the UTXO set).
+|-
+| 11
+| "CTIP" -- Part 2 "Index"*
+| 4
+| hex
+| Of the CTIP, this is second element of the pair: the Index. See #10 above.
+|}
+
+\* Denotes a "convenience field", the entry for this field is derived from other fields, or from the blockchain-state itself. The escrow-creator does not need to provide these values in M1 (or anywhere).
+
+Escrow_DB requires 223 bytes [1+120+32+32+2+32+4] for each escrow in the main blockchain. Of these, 70 bytes [2+32+32+4] are merely for convenience. Therefore, a sidechain is defined (see "M1") by 153 bytes of information.
+
+====Notes on D1====
+
+# D1 will always exist.
+# D1 will always have a unique sort (first by "Escrow Number" (field #1), and second by "Active" (field #2)). There is only ever one (escrow number, Active) pair. 
+# D1 is updated according to M1 and M2 (below).
+# If a new entry is added to D1 with an "Escrow Number" that is already in use, then this entry will either eventually be removed (because it was not supported with an M2), or it will eventually overwrite the old entry (if it *was* supported via M2).
+
+
+====Notes on D1====
+
+=====Obligations Placed on Miners=====
+
+Miners have always upgraded their software according to criteria that are known only to them (in other words, "whenever they want").
+
+However, this soft fork imposes two new criteria upon them. First: miners should only upgrade their software, if any modification to the portfolio of sidechains [that are added/removed in the upgrade] can be expected to increase miner wealth. Trivially, this implies that miners should make sure that the upgrade doesn't overwrite (and destroy) an existing sidechain that they like! But, more seriously, it implies that miners should take an interest in what the sidechain is doing to the mainchain and other sidechains (see below).
+
+===== Destructive Sidechain Interference =====
+
+People frequently emphasize that miners should have "as little control" as possible. It is a very safe claim to make, and a very easy sentence to write. Much harder is to determine exactly what this minimum value is, and how to achieve it. Harder still is to untie the knot of who is actually controlling what, in a decentralized, interacting system.
+
+Certainly, miners can not have "zero control" -- for that is the same as to just remove them from the system altogether. Some rules are enforced "on miners by nodes" (such as the infamous blocksize limit); other rules are enforced by nodes but are narrowly-controlled by miners (such as the proof-of-work itself, or the block's timestamp). Thirdly, some rules are enforced by both against each other (such as the rule against including invalid txns or double-spent txns), for mutual benefit.
```

Comment:
The rules against invalid or double-spent txns are exactly identical to the rules against exceeding the block weight limit. I'm not sure what distinction you are implying exists here...

Reply:
It is a distinction about control, as I say.

If miners had the option to remove the block weight, they probably would. But if they had the option to allow invalid txns, they probably would not.

Reply:
That seems unlikely. Miners could profit a lot if invalid transactions were allowed.

Reply:
I don't see how. It would just make Bitcoin more annoying to use for everyone, with no benefit. So BTC would be less useful, and miners are paid in BTC. So miners would be choosing to pay themselves less.

But nonetheless, this establishes the distinction you asked for. It is about which constraints are protecting whom.

In the future, I and many others believe that miners will switch to monopolist pricing, and self-impose a blocksize/weight limit on each other. So it would be wrong to say that nodes "control" the limit (because they cannot increase it). But it would be wrong to say that miners "control" the limit as it is driven by their collective need to maximize revenues.